### ISSUE TITLE
WCAG mapping for 'Link has non-empty accessible name' c487ae

### ISSUE THREAD

**Issue by dd8** [act-rules/act-rules.github.io#1875]
The mapping for this rule is:

4.1.2 Name, Role, Value (Level A)
2.4.4 Link Purpose (In Context) (Level A)
2.4.9 Link Purpose (Link Only) (Level AAA)

but failed example 9 is also a 1.1.1 failure:

https://act-rules.github.io/rules/c487ae#failed-example-9

This example matches the 1.1.1 failure condition in F65
https://www.w3.org/TR/WCAG20-TECHS/F65.html

**Comment by Jym77** [comment]
The rules maps to SC that are guaranteed to fail if the rule fails, to maintain the fail-on-fail relationship and prevent false positives.

That is, we guarantee that when the rule fails than 2.4.4, 2.4.9 and 4.1.2 also fail.
If we added a 1.1.1 mapping, we would have a lot of false positive where the rule can fail on a page where 1.1.1 is satisfied, but the mapping would still say "if the rule has at least one failed outcome, then 1.1.1 fails", which would be incorrect.

This is why this rule does not map to 1.1.1. It is possible that failing this rule creates a failure of 1.1.1, but it is not guaranteed.

**Comment by dd8** [comment]
That's fine. Just wanted to flag that failing SC 1.1.1 in addition to the other SCs for Example 9 is correct, so the conformance mapping cannot require that only 4.1.2, 2.4.4 and 2.4.9 are flagged for Example 9

**Comment by Jym77** [comment]
That's an awfully good point 🤔 

On one hand, [Understanding ACT Consistency](https://www.w3.org/WAI/standards-guidelines/act/implementations/#understanding-act-consistency) says that the tool should report all the SC that are expected to fail but doesn't mention reporting more SC (for all or some of the test cases).
On the other hand it seems that on some rule [Alfa has a partially consistent mapping](https://www.w3.org/WAI/standards-guidelines/act/implementations/alfa/#breakdown-ffd0e9) only because it reports too many SCs…

But it may indeed be the case that reporting "too many" SCs is actually fine for some (or even all…) failed test cases. And we should probably consider the implementation as consistent in these cases.

I do feel this ties a bit with the discussion we had around removing SCs for some rules (https://github.com/act-rules/act-rules.github.io/pull/1854#pullrequestreview-1001009558, [CG call](https://www.w3.org/2022/06/09-act-r-minutes.html#t05), and https://github.com/act-rules/act-rules.github.io/pull/1862 for the latest occurrence). Because a solution could be to say "this rule must report A.B.C and may report X.Y.Z".

**Comment by WilcoFiers** [comment]
This is different from the interpretation we took when we wrote this rule. I don't think area elements should fail 1.1.1, they aren't the non-text content, they are links on top of non-text content. I think this technique is wrong. We can debate whether or not this was the correct decision, but the intent was to explicitly not fail area elements under 1.1.1. If we want to change that we may need to pull area elements into a separate rule, in the same way that we put image buttons into a separate rule.

I don't think that's the correct interpretation though. If you take a native link with no content, give it an accessible name, and position it on top of an image to simulate the effect of an area element - I don't think that suddenly then becomes non-text content. Similarly, I don't think links inside SVG are non-text content, regardless of if those have `<text>` inside them or not.

**Comment by kengdoj** [comment]
SC 1.1.1 should be mapped as a [secondary requirement](https://w3c.github.io/wcag-act/act-rules-format.html#example-fba1117e). 

This rule was used as Example 9 in the ACT Rules Format draft update.

**Comment by kengdoj** [comment]
Update will be made by https://github.com/act-rules/act-rules.github.io/pull/2060.
